src/components/SignUp/SignUpForm.tsx

```diff
@@ -0,0 +1,170 @@
+
+		if (!nameIsValid || !emailIsValid || !passwordIsValid) {
+			return;
+		}
+		const form = e.currentTarget;
+
+		const formData = new FormData(form);
+		const nameValue = (formData.get('name') as string ?? '').trim();
+		const emailValue = (formData.get('email') as string ?? '').trim();
+		const passwordValue = (formData.get('password') as string ?? '').trim();
+
+		setIsLoading(true);
+		await signup(nameValue, emailValue, passwordValue);
+	};
+
+	return (
+		<form className='login-form' onSubmit={handleSubmit}>
+			<div>
+				<h2 className='center'>Welcome! Let's set up your account.</h2>
+			</div>
+			<div className='input-block'>
+				<label className='block' htmlFor='name-input'>Name</label>
+				<input
+					id='name-input'
+					type='text'
+					name='name'
+					placeholder='Your name'
+					minLength={1}
+					required
+					aria-describedby='name-input-helper-text'
+					ref={nameInputRef}
+				/>
+				<div id='name-input-helper-text'>
+					<span>Please enter a valid name that is at least 1 character long.</span>
+				</div>
+			</div>
+
+			<div className='input-block'>
+				<label className='block' htmlFor='email-input'>
+					E-mail<span>REQUIRED</span>
+				</label>
+				<input
+					id='email-input'
+					type='email'
+					name='email'
+					placeholder='Your account e-mail'
+					required
+					aria-describedby='email-input-helper-text'
+					ref={emailInputRef}
+				/>
+				<div id='email-input-helper-text'>
+					<span>Insert the email you'll use for this account</span>
+				</div>
+			</div>
+
+			<div className='input-block'>
+				<label className='block' htmlFor='password-input'>
+					Password:<span>REQUIRED</span>
+				</label>
+				<input
+					id='password-input'
+					type='password'
+					name='password'
+					value={password}
+					onChange={handleChange}
+					placeholder='Your password'
+					required
+					aria-describedby='password-input-strength password-input-suggestion'
+					ref={passwordInputRef}
+				/>
+				<div className='passwordError' hidden={strength === null}>
+					<div className='text-size'>
+						<div id='password-input-strength' aria-live='polite'>
+							<span>Password strength: {strengthLabels[strength || 0]}</span>
+						</div>
+						<div className='password-meter' data-password-strength={strengthLabels[strength || 0]} aria-hidden='true'>
+							<span className='password-meter-level'></span>
+							<span className='password-meter-level'></span>
+							<span className='password-meter-level'></span>
+						</div>
+						<div id='password-input-suggestion' className='suggestion' data-password-strength={strengthLabels[strength || 0]}>
+							<span className='suggestion-icon' />
+							<p>Suggestions: {feedback}</p>
+						</div>
+					</div>
+				</div>
+			</div>
+
+			<Button type='submit' isLarge={true} style={{ color: 'white', background: '#ED343F' }} aria-label='Complete sign-up form button' disabled={isLoading}>
```

Comment:
```suggestion
			<Button type='submit' isLarge={true} style={{ color: 'white', background: '#ED343F' }} aria-disabled={isLoading}>
```

There is no need for using an `aria-label` here as it is redundant with the button text.

Also: setting the `disabled` attribute _hides_ the button from assistive technologies. Here the `aria-disabled` is better as it does not hide the button and also conveys to assistive technologies that it is disabled.